### ISSUE TITLE
EKS auth issues using exec provider and aws eks get-token

### ISSUE THREAD

**Issue by GenPage** [tomplus/kubernetes_asyncio#203]
Hello, I'm attempting to do some local development with this library and have been unable to auth with my test EKS cluster.

OS: Mac Monterey
Python: 3.9.11
Library version: 22.6.4

I am just using example code to list pods, in fact this issue seems very similar to #188 but I am getting a 401 instead of a 403 and unable to auth with the cluster at all. 

My local kubectl commands auth just fine. I assumed it was an issue with the aws command within the kubeconfig so I manually ran the aws eks get-token command and pasted the token into a test user and the example code worked. I'm using `aws eks update-kubeconfig` to keep my kubeconfig current.

kubeconfig example:
```
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: L*==
    server: https://testcluster-endpoint.us-east-1.eks.amazonaws.com
  name: testcluster
contexts:
- context:
    cluster: testcluster
    namespace: test
    user: testcluster
  name: testcluster
current-context: testcluster
kind: Config
preferences: {}
users:
- name: testclusteruser
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1alpha1
      args:
      - --region
      - us-west-2
      - eks
      - get-token
      - --cluster-name
      - testcluster
      command: aws
      env:
      - name: AWS_PROFILE
        value: testcluster_profile
      provideClusterInfo: false
```

Error log:
```
DEBUG:root:kubeconfig loader - current-context arn:aws:eks:us-east-2:<redacted>:cluster/us-east-2-test-k8s, cluster arn:aws:eks:us-east-2:<redacted>:cluster/us-east-2-test-k8s, user arn:aws:eks:us-east-2:<redacted>:cluster/us-east-2-test-k8s, provider None
DEBUG:root:Try to use exec provider
DEBUG:    2022-05-03 11:24:08 Listing pods with their IPs:
DEBUG:bastion-api:Listing pods with their IPs:
DEBUG:kubernetes_asyncio.client.rest:response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}\n'
```

**Comment by tomplus** [comment]
Strange. No errors from the kubeconfig loader, so it looks like the response from `eks` command is interpreted correctly. I don't have access to eks cluster right now. Could you verify if token is used by printing the value of `client.configuration.api_key['BearerToken'])`? Thanks.

**Comment by GenPage** [comment]
`client.configuration.api_key` threw some errors for me so I had to use `api.configuration.api_key` instead and it does return a token.

```
DEBUG:root:kubeconfig loader - current-context arn:aws:eks:us-east-2:<redacted>:cluster/us-east-2-test-cluster, cluster arn:aws:eks:us-east-2:<redacted>:cluster/us-east-2-test-cluster, user arn:aws:eks:us-east-2:<redacted>:cluster/us-east-2-test-cluster, provider None
DEBUG:root:Try to use exec provider
DEBUG:bastion-api:{'BearerToken': 'Bearer k8s-aws-v1.<redacted>'}
DEBUG:    2022-05-03 22:36:03 Listing pods with their IPs:
DEBUG:bastion-api:Listing pods with their IPs:
DEBUG:kubernetes_asyncio.client.rest:response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}\n'
ERROR:    2022-05-03 22:36:04 (401)
Reason: Unauthorized
HTTP response headers: <CIMultiDictProxy('Audit-Id': '36cfb4b7-51c3-4b41-b957-7f53dae9a9dd', 'Cache-Control': 'no-cache, private', 'Content-Type': 'application/json', 'Date': 'Wed, 04 May 2022 02:36:04 GMT', 'Content-Length': '129')>
HTTP response body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}
```

**Comment by tomplus** [comment]
To check if source of the problem is connected with the issue #188, could you downgrade the library to 21.7.1?

**Comment by GenPage** [comment]
It is still the same error.

```
Installing collected packages: kubernetes_asyncio
  Attempting uninstall: kubernetes_asyncio
    Found existing installation: kubernetes-asyncio 22.6.4
    Uninstalling kubernetes-asyncio-22.6.4:
      Successfully uninstalled kubernetes-asyncio-22.6.4
Successfully installed kubernetes_asyncio-21.7.1
```

```
DEBUG:bastion-api:Listing pods with their IPs:
DEBUG:kubernetes_asyncio.client.rest:response body: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"Unauthorized","reason":"Unauthorized","code":401}\n'
```

**Comment by GenPage** [comment]
@tomplus I apologize but I didn't properly test earlier since I was using a custom user in my kube_config with the direct token pasted.

It does indeed work/auth properly using the exec provider + `aws eks get-token` in 21.7.1

**Comment by tomplus** [comment]
I'm closing it now, it has been fixed in v22.6.1. Feel free to reopen if it doesn't work for you.
